platform/darwin/src/NSExpression+MGLAdditions.h

```diff
@@ -42,6 +90,105 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, readonly) id mgl_jsonExpressionObject;
 
+/**
+ Returns a constant expression containing an `NSString`.
+ 
+ This is equivalent to call `[NSExpression expressionForConstant:]`.
+ 
+ @param string The string constant.
+ */
++ (instancetype)mgl_expressionForString:(nonnull NSString *)string;
+
+/**
+ Returns a constant expression containing an `MGLColor`.
+ 
+ This is equivalent to call `[NSExpression expressionForConstant:]`.
+ 
+ @param color The color constant.
+ */
++ (instancetype)mgl_expressionForColor:(nonnull MGLColor *)color;
+
+/**
+ Returns a constant expression containing an `NSValue`.
+ 
+ This is equivalent to call `[NSExpression expressionForConstant:]`.
+ 
+ @param value The value constant.
+ */
++ (instancetype)mgl_expressionForValue:(nonnull NSValue *)value;
+
++ (instancetype)mgl_expressionForTernaryFunction:(nonnull NSString *)conditionString trueExpression:(nonnull NSExpression *)trueExpression falseExpresssion:(nonnull NSExpression *)falseExpression;
```

Comment:
Let’s turn this into a polyfill for the preexisting `+expressionForConditional:trueExpression:falseExpression:`. It’d be a lot friendlier and safer to require an NSPredicate than a freeform string. (What syntax would the string use, for instance?)

Reply:
One of the forms I tried was using an NSPredicate, it didn't work quite as I expected but I didn't use `expressionForConditional:trueExpression:falseExpression` in the underlaying implementation, let me give it a second try.